### PR TITLE
SplunkPy: better error handling in `splunk-search`

### PR DESCRIPTION
**Pull request by ilappe**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-40336

## Description
currently, in `splunk-search` we create the search by the given `query` arg, and then check if it's have results.
in case there is `fatal` error (invalid syntax of the query for example) we return message that there is no results,
but the `splunk-sdk` `Job` instance contains the indications about the failure but no exception was raised.

in the fix, i check the status in the `Job` instance, and raise `DemistoException` according with the `Job` failure message.

**unit test befor fix:**
image

**unit test after fix:**
image

**playground:**
command to reproduce 
```
!splunk-search query="`notable` | where 1= | head 1"
```

image

### CODE REVIEW

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 1]
```diff
@@ -2424,6 +2424,32 @@ def parse_batch_of_results(current_batch_of_results, max_results_to_add, app):
     return parsed_batch_results, batch_dbot_scores
 
 
+def handle_failed_job(job):
```
Comment (DeanArbel) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2427]: Please change the name to `raise_error_for_failed_job`

**Review by DeanArbel — APPROVED** [review]
Please add a breaking change notification (as this will now return error entries where previously the errors were hidden).
Please work with the tech docs team for an appropriate message.

File: Packs/SplunkPy/ReleaseNotes/3_1_37.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### SplunkPy
+
+- improved error handling in the ***splunk-search*** command.
```
Comment (ShirleyDenkberg) [inline Packs/SplunkPy/ReleaseNotes/3_1_37.md:6]:
```suggestion
- Improved error handling in the ***splunk-search*** command. 
```

**Comment by ShirleyDenkberg** [comment]
@DeanArbel Doc review completed.

**Pull request merged**